pkg/services/alerting/eval_context.go

```diff
@@ -221,7 +221,8 @@ func buildTemplateDataMap(evalMatches []*EvalMatch) map[string]string {
 	for _, match := range evalMatches {
 		for tagName, tagValue := range match.Tags {
 			// skip duplicate values
-			if strings.Contains(result[tagName], tagValue) {
+			match := regexp.MustCompile(fmt.Sprintf("\\b%s\\b", tagValue)).FindString(result[tagName])
```

Comment:
You should escape the value string wrt. special characters as shown below. Additionally, `reVal` should be moved outside of the loops.
```suggestion
			reVal := regexp.MustCompile(fmt.Sprintf(`\b%s\b`, regexp.QuoteMeta(tagValue)))
			match := reVal.FindString(result[tagName])
```

Reply:
Ah yes that's what I was missing - thanks for the reminder. BTW the regex compilation can't be moved outside the loop as it depends on tagValue

Reply:
@wbrowne Argh yes :man_facepalming: